telegram/telegram.go

```diff
@@ -0,0 +1,740 @@
+		err = c.Send("انباری داتشته باشد یا نه؟ (با بله یا خیر پاسخ دهید)")
+		return
+	})
+
+	t.Bot.Handle(&btnElevator, func(c telebot.Context) (err error) {
+		if elevator := session.Filters.Elevator; elevator != nil {
+			err = c.Send("داشتن یا نداشتن آسانسور قبلا مشخص شده است", filterMenu)
+			return
+		}
+		session.State = "setting_elevator"
+		err = c.Send("آسانسور داشته باشد یا نه؟ (با بله یا خیر پاسخ دهید)")
+		return
+	})
+
+	t.Bot.Handle(&btnAdDate, func(c telebot.Context) (err error) {
+		min := session.Filters.StartDate
+		max := session.Filters.EndDate
+		if min != nil || max != nil {
+			err = c.Send("بازه تاریخ ایجاد آگهی قبلا مشخص شده است", filterMenu)
+			return
+		}
+		session.State = "setting_ad_date"
+		err = c.Send("لطفا بازه تاریخ ایجاد آگهی را وارد کنید (مثال: 01-01-1403  01-01-1402)")
+		return
+	})
+
+	t.Bot.Handle(&btnSendFilter, func(c telebot.Context) (err error) {
+		session := models.GetUserSession(c.Chat().ID)
+		gormFilter := repository.NewGormFilter(database.GetInstnace().Db)
+
+		gormFilter.Add(session.Filters)
+		return c.Send("فیلتر شما با موفقیت ثبت شد", userMenu)
+	})
+
+	t.Bot.Handle(&btnBackFilterMenu, func(c telebot.Context) (err error) {
+		session.State = ""
+		t.Bot.Handle(&btnSetFilters, t.handleSetFilters)
+		t.Bot.Handle(&btnShareBookmarks, t.handleShareBookmarks)
+		t.Bot.Handle(&btnGetOutputFile, t.handleGetOutput)
+		t.Bot.Handle(&btnDeleteHistory, t.handleDeleteHistory)
+
+		err = c.Send("به صفحه اصلی بازگشتید", userMenu)
+		return
+	})
+
+	err = c.Send("لطفا یک فیلتر را انتخاب کنید", filterMenu)
+	return
+}
+
+// TODO
+func (t *Telegram) handleShareBookmarks(c telebot.Context) error {
+	return nil
+}
+
+// TODO
+func (t *Telegram) handleGetOutput(c telebot.Context) error {
+	session := models.GetUserSession(c.Chat().ID)
+	getOutputFileMenu.Reply(
+		getOutputFileMenu.Row(btnGetAsZip, btnGetViaEmail),
+		getOutputFileMenu.Row(btnBackFilterMenu),
+	)
+
+	// TODO
+	t.Bot.Handle(&btnGetOutputFile, func(c telebot.Context) (err error) {
+		return
+	})
+
+	t.Bot.Handle(&btnGetViaEmail, func(c telebot.Context) (err error) {
+		session.State = "setting_user_email"
+		err = c.Send("لطفا ایمیل خود را وارد کنید")
+		return
+	})
+
+	t.Bot.Handle(&btnBackGetOutputFileMenu, func(c telebot.Context) (err error) {
+		session.State = ""
+		t.Bot.Handle(&btnSetFilters, t.handleSetFilters)
+		t.Bot.Handle(&btnShareBookmarks, t.handleShareBookmarks)
+		t.Bot.Handle(&btnGetOutputFile, t.handleGetOutput)
+		t.Bot.Handle(&btnDeleteHistory, t.handleDeleteHistory)
+
+		err = c.Send("به صفحه اصلی بازگشتید", userMenu)
+		return
+	})
+	return c.Send("نحوه خروجی گرفتن را انتخاب کنید", getOutputFileMenu)
+}
+
+func (t *Telegram) handleDeleteHistory(c telebot.Context) (err error) {
+	telegram_ID := c.Sender().ID
+	userRepo := repository.NewGormUser(database.GetInstnace().Db)
+	userAdsRepo := repository.NewGormUser_Ad(database.GetInstnace().Db)
+	user, e := userRepo.GetByTelegramId(strconv.Itoa(int(telegram_ID)))
```

Comment:
Remember that for things like Id, which are a summary of something, they should all be written in lowercase or uppercase, and since i is written in uppercase here, d should also be uppercase.